src/views/MapView.js

```diff
@@ -105,26 +154,48 @@ function GeoJSONMarkers() {
 
   useEffect(() => {
     setGeojsonFeatGroup(geojsonFeatGroup.clearLayers());
-
     const layers = new L.GeoJSON(RIVResults, {
       onEachFeature: onEachFeature,
       pointToLayer: function (feature, latlng) {
         // Initial traffic lights for risk value
         if (
-          feature.properties.W_channel == null ||
-          feature.properties.W_channel_depth == null
+          selectedCalculationType == calculationTypeEnums.COMPARE &&
+          feature.properties.VAYLAT == null // If calculation type is compare and VAYLAT is null crosses are used. VAYLAT null refers that point is routeline while only navigationlines have VAYLAT
         ) {
-          return L.circleMarker(latlng, geojsonMarkerOptionsGray);
+          // return L.marker(latlng, { icon: crossIcon });
```

Comment:
kommentoitua koodia

Reply:
I tried some alternatives to this, but none seemed to work as expected. I think this is good for now, if anyone finds a working alternative it probably would be good